src/penn_chime/models.py

```diff
@@ -259,31 +267,56 @@ def sir(
 
 def gen_sir(
     s: float, i: float, r: float, gamma: float, i_day: int, policies: Sequence[Tuple[float, int]]
-) -> Generator[Tuple[int, float, float, float], None, None]:
+):
     """Simulate SIR model forward in time yielding tuples.
     Parameter order has changed to allow multiple (beta, n_days)
     to reflect multiple changing social distancing policies.
     """
     s, i, r = (float(v) for v in (s, i, r))
     n = s + i + r
     d = i_day
+
+    total_days = 1
+    for beta, days in policies:
+        total_days += days
+
+    d_a = np.empty(total_days, "int")
+    s_a = np.empty(total_days, "float")
+    i_a = np.empty(total_days, "float")
+    r_a = np.empty(total_days, "float")
+
+    index = 0
     for beta, n_days in policies:
         for _ in range(n_days):
-            yield d, s, i, r
+            d_a[index] = d
+            s_a[index] = s
+            i_a[index] = i
+            r_a[index] = r
+            index += 1
+
             s, i, r = sir(s, i, r, beta, gamma, n)
             d += 1
-    yield d, s, i, r
 
+    d_a[index] = d
+    s_a[index] = s
+    i_a[index] = i
+    r_a[index] = r
+    return {
+        "day": d_a,
+        "susceptible": s_a,
+        "infected": i_a,
+        "recovered": r_a,
+        "ever_infected": i_a + r_a
+    }
 
-def sim_sir_df(
+
+def sim_sir(
```

Comment:
Remove this function and rename gen_sir -> sim_sir.

Gen sir was named because it returned a generator to be consumed by a dataframe constructor.